### PR TITLE
Description of the PMIx standardization process

### PR DESCRIPTION
**Pull request by jjhursey**
* Ref Issue #181

## Acceptance Process Dates
_The PMIx Community will update this section as the PR progresses through the acceptance process._

 |                  | Date | Link to discussion |
| ---------------- | ---- | ------------------ |
| **Reading**      | May 31, 2019     |     https://github.com/pmix/pmix-standard/pull/183#issuecomment-497746041   Resolve to postpone.            |
| **Reading**   |      |                    |
| **Acceptance**   |      |                    |

### CODE REVIEW

**Comment by jjhursey** [comment]
Please take a look and make sure I captured everything from Issue #181. A couple things that I did not include in this draft:
 * The class that is associated with a newly "Accepted" PR is still under discussion in Issue #179. So that part of the PR will be missing and the PR associated with Issue #179 will fill that in.
 * Concern was raised about a "reasonable" time limit on delaying progress so I left out the following bullet point until we can resolve that issue:
    * Anyone may request a delay in the process, for a reasonable amount of time, until they
   can review and participate in the discussion. This must be requested on the PR before
   the next teleconference.

File: README.md [hunk 1]
```diff
@@ -2,21 +2,122 @@
 
 # PMIx Standard
 
-This repository contains the LaTeX source for the PMIx standard document, and discussions regarding the PMIx standard.
+This repository contains the LaTeX source for the PMIx Standard document, and discussions regarding the PMIx standard.
 
-## Contributing to the PMIx Standard
+Contributions of any form (e.g., use cases, questions, suggestions) are encouraged from anyone interested in the evolution of the PMIx Standard.
+You can participate in the discussion on the [PMIx Forum mailing list](https://groups.google.com/group/pmix-forum),
```
Comment (SteVwonder) [inline README.md]: Minor nit: somewhere I think we decided to only (mostly?) use the mailing list for announcements/broadcasts rather than discussions (which should mostly happen on GitHub via PRs and issues).  Following that should we emphasize PRs/Issues by putting them first in the list?

File: README.md [hunk 2]
```diff
@@ -2,21 +2,122 @@
...
+    * Reference the corresponding Issue(s). Keep the PR marked as [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) until an implementation is available and the PR is ready for broader discussion.
+    * Send an email to the [PMIx Forum mailing list](https://groups.google.com/forum/#!forum/pmix-forum) once the PR is ready for discussion.
+    * Once the discussion has resolved (i.e., dissenting opinions and questions have been addressed) then it can move to an official "Reading" in the next teleconference. Email the mailing list once it is ready for "Reading" to have it added to the teleconference agenda.
+ 3. Official "**Reading**" of the PR during a teleconference
+    * If no revisions are required or objections raised (either on the PR or in the teleconference) then the PR can be considered for "Acceptance" in the next teleconference.
+ 4. Official "**Acceptance**" of the PR during a following teleconference
+    * If no objection is raised (either on the PR on in the teleconference) then the PR is merged into the document and, if also resolved, the corresponding Issue is closed.
```
Comment (SteVwonder) [inline README.md]: `PR on in` -> `PR or in`

File: README.md [hunk 3]
```diff
@@ -2,21 +2,122 @@
...
+
+**General Statements:**
+
+ * Any individual or organization may participate in the discussion on a topic.
+ * Regular teleconferences are announced on the PMIx Forum mailing list
+   - Teleconferences provide a high bandwidth discussion format for various topics
+     surrounding the PMIx Standard, and a synchronization mechanism to guide progress.
+   - Notes from the teleconferences will be archived on the PMIx Forum wiki.
```
Comment (SteVwonder) [inline README.md]: Do we have a link to the wiki?  Is it this: https://github.com/pmix/pmix-standard/wiki#meeting-information?

**Comment by SteVwonder** [comment]
Reading back over #181, we did discuss adding GitHub templates for PRs and Issues.  Do we want to split that out into a separate PR or include it here?

File: README.md (hunk 1 above)
Comment (jjhursey) [reply README.md]: Yeah, we can reword a bit to better emphasize that. The next paragraph really goes into more detail on it. Folks may still post to the mailing list with general procedural questions ("how do I file a clarification?") and we can redirect them to the GitHub site.

Maybe rephrase like:
> You can participate in the discussion on the Pull Requests/Issues on this GitHub repository or on the regular teleconference (announced on the [PMIx Forum mailing list](https://groups.google.com/group/pmix-forum)). The PMIx Forum mailing list is primarily used for announcements and general questions about the PMIx Standard process (e.g., "When is the next release scheduled?" or "I am having trouble filing an issue").

File: README.md (hunk 3 above)
Comment (jjhursey) [reply README.md]: Yeah I'll embed that link.

**Comment by jjhursey** [comment]
@SteVwonder We can do that as separate PRs.
 * I filed PR #187 for Issue templates.
 * I'll create a proposal for the PR template

File: README.md (hunk 2 above)
Comment (jjhursey) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (jjhursey) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -2,21 +2,122 @@
...
 
-### Participatin in the writing
+The PMIx standardization process considers contributions by using the general workflow outlined below:
 
-If you want to write a portion of the standard (or even fix a typo or two), then use the Pull Request mechanism to post the change for review.
+ 1. Open a **GitHub Issue** describing your contribution for general discussion.
+ 2. Open a **GitHub Pull Request (PR)** if a specific change to the PMIx Standard is needed.
+    * Reference the corresponding Issue(s). Keep the PR marked as [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) until an implementation is available and the PR is ready for broader discussion.
+    * Send an email to the [PMIx Forum mailing list](https://groups.google.com/forum/#!forum/pmix-forum) once the PR is ready for discussion.
+    * Once the discussion has resolved (i.e., dissenting opinions and questions have been addressed) then it can move to an official "Reading" in the next teleconference. Email the mailing list once it is ready for "Reading" to have it added to the teleconference agenda.
```
[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: Yeah I can work on a paragraph about the 'lack of dissent' rule.

[the remark this answers is not in the document]
Comment (rhc54) [reply README.md]: I really disagree with this assertion that a single dissenting opinion blocks a proposed change to the standard. Gaining 100% agreement on such changes is virtually impossible and unnecessary.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (rhc54) [reply README.md]: Note that it is illegal to record conversations in some countries except under very specific circumstances, and that many people and organizations object to that practice.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gvallee) [reply README.md]: BTW, I do not personally have a strong opinion on either way. I just would like it to be clarified to avoid future misunderstandings about the intent of the rule.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (gvallee) [reply README.md]: I completely forgot about that and it is a very good point... I know we always as for authorization when we do it and ask people to drop if it is a problem for them, but that may be a too much here.
Feel free to mark my comments about recording as `resolved` and take it into consideration.

File: README.md [hunk 5]
```diff
@@ -2,21 +2,122 @@
...
+ 3. Official "**Reading**" of the PR during a teleconference
+    * If no revisions are required or objections raised (either on the PR or in the teleconference) then the PR can be considered for "Acceptance" in the next teleconference.
```
[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: I see your point. Let me try to tighten that up a bit.

How does the below sound as a replacement for that bullet (and similar bullets throughout the document):
> If no revisions are required and all objections have been addressed then the PR can be considered for "Acceptance" in the next teleconference. Any revisions or objections raised during the teleconference will be noted on the PR during or immediately after the meeting.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: I think that recording opens up a can of worms. It would be nice to have a better archive of the conversation than meeting notes, but I can certainly see folks not participating in the conversation on a call if they know it will be recorded and publicly available.

By forcing the conversation and notes back to the PRs the utility of the meeting notes is somewhat diminished anyway. My hope is that the meeting notes become fairly minimal in the future with just direct references to comments on specific PRs.

[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: Per @gvallee 's comment above I'm going to resolve this conversation. If folks want to bring it back they can re-open it.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: There was a little discussion on this during the [May 10 teleconf](https://github.com/pmix/pmix-standard/wiki/Meeting-2019-05-10).

I think this discussion goes beyond a PR, and needs to be discussed on the Issue.

File: README.md (hunk 1 above)
Comment (jjhursey) [reply README.md]: I just committed the new language. Thanks.

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: I just put in that new language to this bullet and the one just below it.

**Comment by jjhursey** [comment]
I accidentally removed the Draft PR status from this PR. Unfortunately, GH doesn't have the means to put the PR back into the Draft PR state. I think this is ready for review. There are some modification that we know are coming, but those will come in as additional commits to this PR before it moves to **Reading** - which it is not ready for quite yet.

**Comment by jjhursey** [comment]
Notes from May 17, 2019 teleconf:
 * In this PR keep the language a little vague with `dissenting opinions and questions have been addressed`.
 * Followup with discussion in Issue #181 on more specific language. Which will be a separate PR.
 * This PR will be presented for **Reading** on the May 31, 2019 meeting.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jjhursey) [reply README.md]: Per teleconf - this discussion item will be addressed in a later PR. Leaving this PR a little vague in this area allows the community to determine what 'addressed' means to allow a PR to move forward. More specific language will be determined/discussed in a separate PR.

**Comment by jjhursey** [comment]
**Reading** scheduled for the May 31, 2019 teleconference.

Use this comment to collect **Straw Poll** results.
 * You may vote anytime before the scheduled meeting.
 * The poll closes at the scheduled meeting.
 * If you approve the **Reading** of this PR indicate by placing a :+1: emoji on this comment.
 * If you do not approve the **Reading** of this PR indicate by placing a :-1: emoji on this comment. Then follow up with either suggested changes to the PR or an additional comment on the PR describing your concern for discussion.

File: README.md [hunk 6]
```diff
@@ -2,21 +2,131 @@
...
+       Any revisions or objections raised during the teleconference will be noted on the PR during or immediately after the meeting.
+ * The PR may be voluntarily closed as "**Withdrawn**" by the corresponding author(s) for any reason. For example, if a dissenting opinion is raised that cannot be resolved in the scope of that PR.
+ * If revisions are required or further discussion is needed then the interested parties should continue discussion on the PR.
+   - If no changes to the PR result from the discussion then the PR can be considered for "**Acceptance**" without the need for another "**Reading**".
+   - If changes are made to the PR (with reasonable exceptions for minor spelling/grammar changes) then another "**Reading**" must be scheduled (See Phase 2) to allow interested parties to review the changes. This will include another "**Straw Poll**".
+   - If major changes are required to the PR the author(s) should consider a separate, replacement PR, and marking the original PR as "**Withdrawn**".
+ * If no dissent is voiced then the PR can be considered for "**Acceptance**" as early as the next teleconference.
```
Comment (abouteiller) [inline README.md]: This requirement is too stringent. This can be understood as veto power, which would make any progress impossible.

Comment (jjhursey) [reply README.md]: Maybe we can reword to:
> If all dissenting opinions and questions have been addressed then the PR can be considered for "**Acceptance**" as early as the next teleconference.

Leaving the definition of `addressed` open. We will need to more crisply define that.

**Comment by jjhursey** [comment]
Per teleconf:
 * Decided to delay reading until Issue #179 is resolved.

**Comment by jjhursey** [comment]
Closing in favor of PR #193
